### PR TITLE
i18n implementation

### PR DESCRIPTION
**Pull request by schraj**
Implements internationalization (i18n) using the `react-i18next` library. Carries forward the translation files from the Choo front-end version. This work is done in support of issue #10.

### CODE REVIEW

File: src/services/i18nTesting.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// initialize i18n cache
```
Comment (cdoremus) [inline src/services/i18nTesting.ts]: I think that this file should reside in a top-level test folder so it doesn't end up deployed to production. I know that the prod deployment ignores *.test.* files, but I'm not sure about this one.

File: package.json [hunk 2]
```diff
@@ -3,14 +3,19 @@
   "version": "0.1.0",
```
Comment (cdoremus) [inline package.json:3]: All the @types/* files should be in the devDependencies section. Also, you should `yarn add -D @types/react-i18next` to pull in the type definition for that library.

File: src/setupTests.js [hunk 3]
```diff
@@ -0,0 +1 @@
+// add logic here that needs to be run before any tests are run.
```
Comment (cdoremus) [inline src/setupTests.js]: This file should move to a top-level `test` folder and the reference in `scripts/test.js` should me modified.

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (cdoremus) [inline .vscode/launch.json]: Is there a reason this file should be in the repository? Although vscode is probably the best editor to use to do this work, we should not require it.

File: src/components/call/Call.test.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import i18n from '../../services/i18nTesting';
+import { Call } from './index';
+import { Issue, DefaultIssue } from '../../common/model';
+import { CallState } from '../../redux/callState';
+
+test('Call component should call info if it is defined', () => {
```
Comment (cdoremus) [inline src/components/call/Call.test.tsx]: Should this be a snapshot test? Also, the test description needs work.

File: src/components/location/Location.test.tsx [hunk 6]
```diff
@@ -1,70 +1,79 @@
...
-  const zip = '10001';
-  const form = component.find('form').first();
-  // console.log('node: \n', form.debug());
-  form.simulate('submit', { preventDefault: jest.fn(), target: {elements: {address: { value: zip } }}} );
-  expect(setLocation).toBeCalledWith(zip);
-});
+//   const setLocation = jest.fn();
+//   const clearLocation = jest.fn();
 
-test('Location component clearLocation() should be called upon submit if isLoading=false', () => {
```
Comment (cdoremus) [inline src/components/location/Location.test.tsx]: This test should be uncommented. It runs successfully.

File: src/components/location/Location.test.tsx [hunk 7]
```diff
@@ -1,70 +1,79 @@
...
-      isLoading={isLoading}
-      setLocation={setLocation}
-      clearLocation={clearLocation}
-    />
-  );
-  const button = component.find('button').first();
-  button.simulate('click', { preventDefault: jest.fn()} );
-  expect(clearLocation).toBeCalled();
-});
+// test('Location component clearLocation() should be called upon submit if isLoading=false', () => {
```
Comment (cdoremus) [inline src/components/location/Location.test.tsx]: This test should be uncommented. It runs successfully.

**Review by cdoremus — COMMENTED** [review]
Good job getting i18n with all tests to run!

As I mentioned to you on Slack, the following files have compile errors, mostly due to a missing 't' attribute in component elements:
Sidebar, AboutPage, Done, DonePage, Call, CallPage, Why5Calls, HomePage, Footer, Layout

File: src/setupTests.js (hunk 3 above)
Comment (schraj) [reply src/setupTests.js]: removed it

File: src/services/i18nTesting.ts (hunk 1 above)
Comment (schraj) [reply src/services/i18nTesting.ts]: getting a TS error when it is outside of the src folder.  I'm thinking we could add another convention like the ".test.ts" excludes to exclude ".fixture.ts" from the prod build.  thoughts?

File: .vscode/launch.json (hunk 4 above)
Comment (schraj) [reply .vscode/launch.json]: removing it

File: package.json (hunk 2 above)
Comment (schraj) [reply package.json:3]: fixed. didn't realize yarn had a different syntax

**Comment by schraj** [comment]
I think I fixed all the items.  It turns out that Jest tests work fine with the regular i18n setup.  Since I'm not mocking it, but was just creating a locally loaded instance, I don't think we need the test i18n instantiation file anymore.  

It isn't "pure" unit testing to inject the i18n instance instead of mocking it.  However, I'm not sure that it is a big issue.  It could be a task in the future to work it out.  I didn't mock it in unit tests in the old site either.

**Review by cdoremus — COMMENTED** [review]
Looks good now. If you are not using the `localization` constant in `src/common/constants.ts`, please remove it from the file.

A few lint warnings that popped up when starting the app should also be fixed:
```
Compiled with warnings.

./src/redux/remoteData/asyncActionCreator.ts
[2, 9]: missing whitespace
[3, 25]: missing whitespace
[3, 9]: missing whitespace
./src/services/apiServices.ts
[1, 29]: missing whitespace
[1, 16]: missing whitespace
```
The tests all passed and they ran in 2-3 seconds, so I have no problem with running the i18n tests without a mock.

**Pull request merged**